test_client/src/main/scala/io/reactivesocket/tck/ClientDSL.scala

```diff
@@ -4,7 +4,7 @@ import java.io.{File, PrintWriter}
 
 class ClientDSL {
 
-  val writer: PrintWriter = new PrintWriter(new File(this.getClass.getSimpleName + ".txt"))
+  var writer: PrintWriter = new PrintWriter(new File(this.getClass.getSimpleName + ".txt"))
```

Comment:
feels very "iffy". when/why/who would set this value (I've seen below, yeah the class extending it, but then - why not as constructor parameter, which would be more understandable)